lib/request.coffee

```diff
@@ -322,4 +306,53 @@ module.exports = getRequest = ({
 	# (or resolve to) a new response, or throw/reject.
 	###
 
+	###*
+	# @summary Refresh token on user request
+	# @function
+	# @public
+	#
+	# @description
+	# This function automatically refresh token on user request.
+	# The token is refreshed even if it is not expired yet.
+	# Error is thrown if user is not authorized.
+	#
+	# @param {String} options.url - relative url
+	#
+	# @returns {String} token - new token
+	#
+	# @example
+	# request.refreshToken
+	# 	baseUrl: 'https://api.resin.io'
+	###
+
+	exports.refreshToken = (options = {}) ->
+
+		{baseUrl} = options
+
+		# Only refresh if we have resin-auth
+		return if not (auth?)
```

Comment:
I see that for normal requests we currently skip token handling if auth isn't present (though tbh, I have no idea when that happens).

In this case though, I think we should throw. If auth isn't present, and you've asked us to refresh your token, you're not going to get the result you expect. Just a generic `new Error` with some sensible message is fine.